indico/modules/auth/controllers.py

```diff
@@ -200,6 +205,60 @@ def _create_user(self, data, handler):
         return handler.redirect_success()
 
 
+class RHUserAccounts(RHUserBase):
+    """Displays user accounts"""
+
+    def _create_form(self):
+        if self.user.local_identity:
+            defaults = FormDefaults(username=self.user.local_identity.identifier)
+            local_account_form = LocalLoginEditForm(identity=self.user.local_identity, obj=defaults)
+        else:
+            local_account_form = LocalLoginAddForm()
+        return local_account_form
+
+    def _handle_add_local_account(self, form):
+        identity = Identity(provider='indico', identifier=form.data['username'], password=form.data['password'])
+        self.user.identities.add(identity)
+        flash(_("Local account added successfully"), 'success')
+
+    def _handle_edit_local_account(self, form):
+        self.user.local_identity.identifier = form.data['username']
+        self.user.local_identity.password = form.data['password']
```

Comment:
how would the form handle it? you probably get a None if it doesn't validate, but then you end up setting the password to None which should raise a `ValueError('Password may not be empty')` (see indico/util/passwords.py)